### PR TITLE
Make email.message.Message generic over the header type

### PR DESCRIPTION
**Pull request by srittau**

### CODE REVIEW

**Comment by srittau** [comment]
The email/http `Message`/`Header` class hierarchy is quite "creative".

File: stdlib/email/message.pyi [hunk 1]
```diff
@@ -72,22 +72,22 @@ class Message:
     # Morally, the return type should be `AnyOf[_HeaderType, None]`,
     # which we could spell as `_HeaderType | Any`,
     # *but* `_HeaderType` itself is currently an alias to `Any`...
-    def __getitem__(self, name: str) -> _HeaderType: ...
-    def __setitem__(self, name: str, val: _HeaderTypeParam) -> None: ...
+    def __getitem__(self, name: str) -> _HeaderT: ...
```
Comment (Avasam) [inline stdlib/email/message.pyi:75]: According to the above comment, this should maybe be using "the Any trick"?

```suggestion
    def __getitem__(self, name: str) -> _HeaderT | Any: ...
```

File: stdlib/email/message.pyi [hunk 2]
```diff
@@ -72,22 +72,22 @@ class Message:
     # Morally, the return type should be `AnyOf[_HeaderType, None]`,
     # which we could spell as `_HeaderType | Any`,
     # *but* `_HeaderType` itself is currently an alias to `Any`...
```
Comment (Avasam) [inline stdlib/email/message.pyi:74]: (see other comment, github wouldn't let me do a suggestion across a deleted line)
```suggestion
```
```suggestion
    # so using "the Any trick" instead.
```
On that note, we never ended up making an alias for it, something like `MaybeNone`. https://github.com/python/typeshed/issues/11094

**Review by Avasam — CHANGES_REQUESTED** [review]
A small adjustment for a comment mentioning the Any trick, otherwise this looks good to me.

**Review by Avasam — APPROVED** [review]

**Pull request merged**